fsspec/implementations/http.py

```diff
@@ -104,10 +108,6 @@ def __init__(
         # TODO: Maybe rename `self.kwargs` to `self.request_options` to make
         #       it clearer.
         request_options = copy(storage_options)
-        self.use_listings_cache = request_options.pop("use_listings_cache", False)
-        request_options.pop("listings_expiry_time", None)
-        request_options.pop("max_paths", None)
-        request_options.pop("skip_instance_cache", None)
```

Comment:
What about those args? I think they are not needed/used anymore.

Reply:
We need to make sure not to leak any kwargs to downstream implementations, since some of them take any kwargs and assume they are to be propagated to, for example, the http session constructor. This could in principle be handled in `_Cached.__call__`, which happens before any implementation-specific `__init__`.

Reply:
If we use the listings_cache_options specifically, they are only passed to the _DirCache in `AbstractFileSystem` class but do not go further down the hierarchy.